### PR TITLE
Use try with resources when creating AdminClient

### LINKED ISSUE #660: Too many open files using spring cloud stream 2.1 + Kafka
After upgrading to spring cloud stream 2.1 on a project using multi binder kafka(2 brokers) and rabbit(1 broker) we are facing too many open files problem.

The number of files opened keeps growing evry second until reaching the limit defined by the os(redhat 7.3).

I use the actuator endpoint to monitor: /actuator/metrics/process.files.open

```
{
"name": "process.files.open",
"description": "The open file descriptor count",
"baseUnit": "files",
"measurements": [
{
"statistic": "VALUE",
"value": 1686
}

]
}
```

using cmd> lsof -p pidOfProcessJavaApplication
I see a lot of TCP connection established related to brokers kafka

Using the version 2.0 of spring cloud stream the problem is not observed, the number of open files is stable around 80 files. 
There's no problem using the exact following versions:

```
 +- org.springframework.cloud:spring-cloud-stream-binder-kafka-core:jar:2.0.0.RELEASE:compile
[INFO] |  |  \- org.springframework.integration:spring-integration-kafka:jar:3.0.3.RELEASE:compile
[INFO] |  +- org.apache.kafka:kafka-clients:jar:1.0.2:compile
[INFO] |  |  +- org.lz4:lz4-java:jar:1.4:compile
[INFO] |  |  \- org.xerial.snappy:snappy-java:jar:1.1.4:compile
[INFO] |  \- org.springframework.kafka:spring-kafka:jar:2.1.10.RELEASE:compile
I'm suspecting that the upgrade to kafka-client 2.0.0 is the potential issue.
```

In a first attempt I wanted to try the spring cloud stream 2.1 with kafka-clients 1.0.2, according to docs it's possible but I'm facing an issue. Here's my maven configuration with exclusion:

```
<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-stream-binder-kafka</artifactId>
    <exclusions>
        <exclusion>
            <groupId>org.apache.kafka</groupId>
            <artifactId>kafka-clients</artifactId>
        </exclusion>
        <exclusion>
            <groupId>org.springframework.kafka</groupId>
            <artifactId>spring-kafka</artifactId>
        </exclusion>
        <exclusion>
            <groupId>org.springframework.integration</groupId>
            <artifactId>spring-integration-kafka</artifactId>
        </exclusion>
    </exclusions>
</dependency>
<dependency>
    <groupId>org.apache.kafka</groupId>
    <artifactId>kafka-clients</artifactId>
    <version>1.0.2</version>
</dependency>
<dependency>
    <groupId>org.springframework.kafka</groupId>
    <artifactId>spring-kafka</artifactId>
    <version>2.2.5.RELEASE</version>
</dependency>
<dependency>
    <groupId>org.springframework.integration</groupId>
    <artifactId>spring-integration-kafka</artifactId>
    <version>3.1.0.RELEASE</version>
</dependency>
```
I'm having the following error:

```
java.lang.NoSuchMethodError: org.apache.kafka.clients.consumer.Consumer.poll(Ljava/time/Duration;)Lorg/apache/kafka/clients/consumer/ConsumerRecords;
    at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.pollAndInvoke(KafkaMessageListenerContainer.java:741)
    at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.run(KafkaMessageListenerContainer.java:698)
    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
    at java.base/java.lang.Thread.run(Thread.java:834)
```

It seems that the problem is not related to spring-kafka because using @kafkalistener with same servers and number of topics the number of opened files does not increase(approximately same using spring cloud stream 2.0)

Any idea about too many open files and how to diagnose? 

Thanks for you help.

### PR DESCRIPTION
**Pull request by iguissouma**
Use try with resources when creating AdminClient to release all associated resources.
Fixes #660

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@iguissouma Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-stream-binder-kafka&pullRequestId=661)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-stream-binder-kafka&pullRequestId=661) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@iguissouma Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-stream-binder-kafka&pullRequestId=661)!

File: spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java [hunk 1]
```diff
@@ -466,11 +466,11 @@ else if (tolerateLowerPartitionsOnBroker) {
 				// In some cases, the above partition query may not throw an UnknownTopic..Exception for various reasons.
 				// For that, we are forcing another query to ensure that the topic is present on the server.
 				if (CollectionUtils.isEmpty(partitions)) {
-					final AdminClient adminClient = AdminClient
-							.create(this.adminClientProperties);
-					final DescribeTopicsResult describeTopicsResult = adminClient
+					try (AdminClient adminClient = AdminClient
+							.create(this.adminClientProperties)){
```
Comment (garyrussell) [inline spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java]: Thanks for the contribution.

Please run `./mvnw clean package` before submitting a PR; you'll see there are style violations (missing space before `{`; trailing spaces on line 473).

Comment (garyrussell) [reply spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java]: I will fix these when merging this time; thanks again.

**Comment by garyrussell** [comment]
Merged as 4ab6432f237ae8c9cbe0133bb80b81a60485150a; cherry-picked to 2.2.x as 683e3f96ba5d35c5aed0a29e9cb7a8099acef08b.

**Pull request closed without merging**